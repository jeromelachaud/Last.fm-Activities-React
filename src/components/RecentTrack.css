/* RecentTrack */
.RecentTrack {
  margin-bottom: 5rem;
}

.RecentTrack a:link,
.RecentTrack a:visited {
  color: #111111;
  display: block;
  background-color: rgba(255, 255, 255, 0.8);
}

.RecentTrack a:active {
  text-decoration: none;
}

.RecentTrack a:hover {
  background-color: rgba(255, 255, 255, 1);
  -webkit-transition: background-color .3s ease-in-out;
  transition: background-color .3s ease-in-out;
}

.RecentTrack__info {
  padding: 1rem 0 0.7rem 0.7rem;
  width: 300px;
}

.RecentTrack__date {
  font-size: 80%;
}
