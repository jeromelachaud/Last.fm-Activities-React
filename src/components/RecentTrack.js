import React from 'react';
import styled from 'styled-components';
import lastfm_placeholder from '../assets/cover-album-placeholder.svg';

const RecentTrackWrapper  = styled.ul`
  margin-bottom: 2.5rem;
`;

const RecentTrackItem = styled.li`
  display:block;

  a {
    display: inline-block;

    &:link,
    &:visited {
      color: #111111;
      background-color: rgba(255, 255, 255, 0.8);
    }

    &:hover {
      background-color: rgba(255, 255, 255, 1);
      -webkit-transition: background-color .3s ease-in-out;
      transition: background-color .3s ease-in-out;
    }

    @media screen and (min-width:1024px) {
      display: inline-flex;
    }
  }

`;

const RecentTrackInfo = styled.ul`
  padding: .75rem 1rem;
  width: 300px;
  line-height: 1.5;
`;
const RecentTrackDate = styled.li`
  font-size: 80%;
`;

let RecentTrack = React.createClass ({

  propTypes: {
    recentTrack: React.PropTypes.object.isRequired
  },

  render() {
    const {
      recentTrack
    } = this.props;

    const {
      artist,
      album,
      date,
      image,
      name,
      url
    } = recentTrack;

    const imageApiResponse = image['3']['#text'];

    let dateElement;
    if (date) {
      dateElement = (
        <RecentTrackDate>{date['#text']}</RecentTrackDate>
      );
    }
    let imageSource;
    if (imageApiResponse.length !== 0) {
      imageSource = (imageApiResponse);
    } else {
      imageSource = (lastfm_placeholder);
    }

    return (
      <RecentTrackWrapper>
        <RecentTrackItem>
          <a href={url}>
            <img
              src={imageSource}
              alt={name} />
            <RecentTrackInfo>
              <li><strong>{artist['#text']}</strong></li>
              <li>{name}</li>
              <li><em>{album['#text']}</em></li>
              {dateElement}
            </RecentTrackInfo>
          </a>
        </RecentTrackItem>
      </RecentTrackWrapper>
    );
  }
});

export default RecentTrack;
