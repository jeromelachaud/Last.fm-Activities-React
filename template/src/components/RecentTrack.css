/* List */
.RecentTrack__item {
  margin-bottom: 5rem;
}

.RecentTrack__item a:link,
.RecentTrack__item a:visited {
  color: #111111;
  display: block;
  background-color: rgba(255, 255, 255, 0.8);
}

.RecentTrack__item a:active {
  color: lime;
  text-decoration: none;
}

.RecentTrack__item a:hover {
  background-color: rgba(255, 255, 255, 1);
  -webkit-transition: background-color .3s ease-in-out;
  transition: background-color .3s ease-in-out;
}

.RecentTrack__info {
  padding: 1rem 0 0.7rem 0.7rem;
  width: 300px;
}

.date {
  font-size: 80%; }
