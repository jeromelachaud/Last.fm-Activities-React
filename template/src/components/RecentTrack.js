import React from 'react';
import './RecentTrack.css';

let RecentTrack = React.createClass ({

  render() {
    const {
      recentTrack
    } = this.props;

    const {
      artist,
      album,
      date,
      image
    } = recentTrack;

    let dateElement;
    if (date) {
      dateElement = (
        <li className="RecentTrack__date">{date['#text']}</li>
      );
    }

    return (
      <ul
        className="RecentTrack">
        <li
          className="RecentTrack_item">
          <a href={recentTrack.url}>
            <img src={image['3']['#text']} />
            <ul
              className="RecentTrack__info">
              <li><strong>{artist['#text']}</strong></li>
              <li>{recentTrack.name}</li>
              <li><em>{album['#text']}</em></li>
              {dateElement}
            </ul>
          </a>
        </li>
      </ul>
    );
  }
});

export default RecentTrack;
