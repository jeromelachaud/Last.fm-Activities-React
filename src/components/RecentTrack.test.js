import React from 'React';
import {
  shallow
} from 'enzyme';
import toJson from 'enzyme-to-json';
import RecentTrack from './RecentTrack';

const recentTrack = {
  'name': 'a name',
  'image': [
    {'#text' : 'album'},
    {'#text' : 'album'},
    {'#text' : 'album'},
    {'#text' : 'album'}
  ],
  'artist': ['#text'],
  'album': ['#text'],
  'date': ['#test']
};

describe('<RecentTrack /> componemts', () => {
  const wrapper = shallow(<RecentTrack recentTrack={recentTrack}/>);

  it('should render as expected', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
