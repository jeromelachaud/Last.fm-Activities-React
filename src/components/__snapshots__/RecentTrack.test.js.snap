exports[`<RecentTrack /> componemts should render as expected 1`] = `
<styled.ul>
  <styled.li>
    <a>
      <img
        alt="a name"
        src="album" />
      <styled.ul>
        <li>
          <strong />
        </li>
        <li>
          a name
        </li>
        <li>
          <em />
        </li>
        <styled.li />
      </styled.ul>
    </a>
  </styled.li>
</styled.ul>
`;
